tests/serialization/test_dag_serialization.py

```diff
@@ -1366,6 +1366,88 @@ def _check_taskgroup_children(
             ["group1.group2.teardown2"],
         )
 
+    def test_task_group_setup_teardown_taskgroups(self):
+        """
+        Test TaskGroup setup and teardown taskgroup serialization/deserialization.
+        """
+        from airflow.decorators import setup, task_group, teardown
+        from airflow.operators.empty import EmptyOperator
+
+        execution_date = datetime(2020, 1, 1)
+        with DAG("test_task_group_setup_teardown_task_groups", start_date=execution_date) as dag:
+
+            @setup
+            @task_group
+            def setup():
```

Comment:
probably not a good idea to redefine setup here!  this bit us before :)

Reply:
Oops 🤦‍♂️. Not the last time I'll do that most likely. Thanks.